### PR TITLE
Add documentation for config.yaml

### PR DESCRIPTION
**Pull request by melihyilmaz**
Addressing #29 , added comments to `config.yaml` and added a short explanation about the config file to read the docs. Indicated which parameters can be changed when running inference with Casanovo.

### CODE REVIEW

File: casanovo/config.yaml [hunk 1]
```diff
@@ -1,30 +1,50 @@
...
+# Parameters that can be modified when running inference with Casanovo,
+# i.e. denovo and eval modes in CLI, are marked with "(I)". Other parameters
+# shouldn't be changed unless a new Casanovo model is being trained.
 ###
 
 # Random seed to ensure reproducible results.
 random_seed: 454
 
 # Spectrum processing options.
+# Max number of peaks in spectrum
```
Comment (wsnoble) [inline casanovo/config.yaml:13]: Clarify that peaks are sorted by intensity, and the least intense peaks beyond this number are discarded.

File: casanovo/config.yaml [hunk 2]
```diff
@@ -1,30 +1,50 @@
...
 n_peaks: 150
+# Min peak m/z allowed
```
Comment (wsnoble) [inline casanovo/config.yaml:15]: Clarify that spectra with too few peaks are skipped.

File: casanovo/config.yaml [hunk 3]
```diff
@@ -1,30 +1,50 @@
...
 min_mz: 50.0
+# Max peak m/z allowed
```
Comment (wsnoble) [inline casanovo/config.yaml:17]: Clarify that peaks with m/z values above this are discarded.

File: casanovo/config.yaml [hunk 4]
```diff
@@ -1,30 +1,50 @@
...
 max_mz: 2500.0
+# Min peak intensity allowed
```
Comment (wsnoble) [inline casanovo/config.yaml:19]: Clarify that peaks with intensity below this value are discarded.

File: casanovo/config.yaml [hunk 5]
```diff
@@ -1,30 +1,50 @@
...
 min_intensity: 0.01
+# m/z tolerance to use when removing the precursor peak
```
Comment (wsnoble) [inline casanovo/config.yaml:21]: Clarify that this is +/- (i.e., for 2 Da it's a window of size 4 Da).

File: casanovo/config.yaml [hunk 6]
```diff
@@ -1,30 +1,50 @@
...
 remove_precursor_tol: 2.0  # Da
+# Max precursor charge allowed
```
Comment (wsnoble) [inline casanovo/config.yaml:23]: Clarify that spectra with larger associated charges are skipped.

File: casanovo/config.yaml [hunk 7]
```diff
@@ -1,30 +1,50 @@
...
 max_charge: 10
+# Tolerance for precursor m/z filtering (I)
```
Comment (wsnoble) [inline casanovo/config.yaml:25]: Explain this in more detail.

File: casanovo/config.yaml [hunk 8]
```diff
@@ -1,30 +1,50 @@
...
 precursor_mass_tol: 50  # ppm
+# Isotope error range for precursor m/z filtering (I)
```
Comment (wsnoble) [inline casanovo/config.yaml:27]: Explain this in more detail.

File: casanovo/config.yaml [hunk 9]
```diff
@@ -1,30 +1,50 @@
...
+# Number of attention heads
 n_head: 8
+# Dimensionality of fully connected layers
 dim_feedforward: 1024
+# Number of transformer layers in spectrum encoder and peptide decoder
 n_layers: 9
+# Dropout rate for model weights
 dropout: 0.0
+# Number of dimensions to use for encoding peak intensity
 dim_intensity:
```
Comment (wsnoble) [inline casanovo/config.yaml:42]: Why is there no value here? (and next line as well)

File: casanovo/config.yaml [hunk 10]
```diff
@@ -55,25 +75,39 @@ residues:
   "+43.006": 43.005814      # Carbamylation
   "-17.027": -17.026549     # NH3 loss
   "+43.006-17.027": 25.980265
```
Comment (wsnoble) [inline casanovo/config.yaml:77]: Add comment explaining this.

File: casanovo/config.yaml [hunk 11]
```diff
@@ -1,30 +1,50 @@
 ###
 # Casanovo configuration.
-# Blank entries are interpreted as "None"
+# Blank entries are interpreted as "None".
+# Parameters that can be modified when running inference with Casanovo,
+# i.e. denovo and eval modes in CLI, are marked with "(I)". Other parameters
```
Comment (wsnoble) [inline casanovo/config.yaml:5]: write out "command line interface"

File: docs/getting_started.md [hunk 12]
```diff
@@ -59,7 +59,7 @@ casanovo --help
 ```
 
 All auxiliary data, model, and training-related parameters can be specified in a user created `.yaml` configuration file.
-See [`casanovo/config.yaml`](https://github.com/Noble-Lab/casanovo/blob/main/casanovo/config.yaml) for the default configuration that was used to obtain the reported results.
+See [`casanovo/config.yaml`](https://github.com/Noble-Lab/casanovo/blob/main/casanovo/config.yaml) for the default configuration that was used to obtain the reported results. You can change some of the parameters on this file when de novo sequencing as needed (marked as such) and should not change other parameters unless you're training a new Casanovo model.
```
Comment (wsnoble) [inline docs/getting_started.md:62]: When running Casanovo in eval and denovo mode, you can change some of the parameters in this file, indicated with "(I)" in the file. You should not change other parameters unless you are training a new Casanovo model.

**Review by wsnoble — CHANGES_REQUESTED** [review]

File: casanovo/config.yaml (hunk 2 above)
Comment (melihyilmaz) [reply casanovo/config.yaml:15]: We don't have a minimum number of peaks variable, spectra are skipped only if no peaks are left after pre-processing.

File: casanovo/config.yaml (hunk 9 above)
Comment (melihyilmaz) [reply casanovo/config.yaml:42]: Blank entries are interpreted as "None" as mentioned at the beginning, edited comments to make default behavior clearer.

File: casanovo/config.yaml [hunk 13]
```diff
@@ -1,30 +1,54 @@
...
+# Max precursor charge allowed, spectra with larger charge are skipped
 max_charge: 10
+# Max absolute difference allowed with respect to observed precursor m/z (I)
+# Predictions outside the tolerance range are assinged a negative peptide score
 precursor_mass_tol: 50  # ppm
+# Isotopes to consider when comparing predicted and observed precursor m/z's (I)
 isotope_error_range: [0, 1]
 
 # Model architecture options.
+# Dimentionality of latent representations, i.e. peak embeddings
```
Comment (wsnoble) [inline casanovo/config.yaml:33]: Dimensionality

File: docs/getting_started.md [hunk 14]
```diff
@@ -59,7 +59,7 @@ casanovo --help
 ```
 
 All auxiliary data, model, and training-related parameters can be specified in a user created `.yaml` configuration file.
-See [`casanovo/config.yaml`](https://github.com/Noble-Lab/casanovo/blob/main/casanovo/config.yaml) for the default configuration that was used to obtain the reported results.
+See [`casanovo/config.yaml`](https://github.com/Noble-Lab/casanovo/blob/main/casanovo/config.yaml) for the default configuration that was used to obtain the reported results. When running Casanovo in eval and denovo mode, you can change some of the parameters in this file, indicated with "(I)" in the file. You should not change other parameters unless you are training a new Casanovo model.
```
Comment (wsnoble) [inline docs/getting_started.md:62]: "eval or denovo mode"

**Review by wsnoble — APPROVED** [review]
Just a few typos.

**Pull request merged**